### PR TITLE
Gardener nightly integration job

### PR DESCRIPTION
**Pull request by clebs**
**Description**

Changes proposed in this pull request:

- Add new gardener nightly integration job

**Related issue(s)**
#1757

### CODE REVIEW

File: prow/scripts/cluster-integration/helpers/deprovision-gardener-cluster.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# - GARDENER_CREDENTIALS
+#
+# REQUIREMENTS:
+# - kubectl
+###
+
+readonly NAMESPACE="garden-${GARDENER_PROJECT_NAME}"
+
+kubectl --kubeconfig "${GARDENER_CREDENTIALS}" -n "${NAMESPACE}" annotate shoot "${GARDENER_CLUSTER_NAME}" confirmation.garden.sapcloud.io/deletion=true --overwrite
+kubectl --kubeconfig "${GARDENER_CREDENTIALS}" -n "${NAMESPACE}" delete   shoot "${GARDENER_CLUSTER_NAME}"
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/helpers/deprovision-gardener-cluster.sh]:
```suggestion
kubectl --kubeconfig "${GARDENER_CREDENTIALS}" -n "${NAMESPACE}" delete shoot "${GARDENER_CLUSTER_NAME}"
```

File: prow/scripts/cluster-integration/helpers/kyma-cli.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+install::kyma_cli() {
+    mkdir -p "${INSTALL_DIR}/bin"
+    export PATH="${INSTALL_DIR}/bin:${PATH}"
+    os=$(host::os)
+
+    pushd "${INSTALL_DIR}/bin"
+
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/helpers/kyma-cli.sh]: please delete the extra space

File: prow/scripts/cluster-integration/helpers/kyma-cli.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    Linux)
+      host_os=linux
+      ;;
+    *)
+      log::error "Unsupported host OS. Must be Linux or Mac OS X."
+      exit 1
+      ;;
+  esac
+  echo "${host_os}"
+}
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/helpers/kyma-cli.sh]: new line

File: prow/scripts/cluster-integration/kyma-gardener-integration.sh [hunk 4]
```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env bash
+
+#Description: Kyma CLI Integration plan on GKE. This scripts implements a pipeline that consists of many steps. The purpose is to install and test Kyma using the CLI on real GKE cluster.
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/kyma-gardener-integration.sh]: Please update the description

File: prow/scripts/cluster-integration/kyma-gardener-integration.sh [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+### Cluster name must be less than 20 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# Local variables
+# DNS_SUBDOMAIN="${COMMON_NAME}"
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+
+
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/kyma-gardener-integration.sh]: unnecessary extra space

File: prow/scripts/cluster-integration/kyma-gardener-integration.sh [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+# shout "Generate self-signed certificate"
+# date
+# DOMAIN="${DNS_SUBDOMAIN}.${DNS_DOMAIN%?}"
+# export DOMAIN
+# CERT_KEY=$("${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/generate-self-signed-cert.sh")
+# TLS_CERT=$(echo "${CERT_KEY}" | head -1)
+# TLS_KEY=$(echo "${CERT_KEY}" | tail -1)
+
+
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/kyma-gardener-integration.sh]: unnecessary extra space

File: prow/scripts/cluster-integration/helpers/deprovision-gardener-cluster.sh (hunk 1 above)
Comment (clebs) [reply prow/scripts/cluster-integration/helpers/deprovision-gardener-cluster.sh]: Done thanks!

File: prow/scripts/cluster-integration/helpers/kyma-cli.sh (hunk 3 above)
Comment (clebs) [reply prow/scripts/cluster-integration/helpers/kyma-cli.sh]: Done thanks!

File: prow/scripts/cluster-integration/kyma-gardener-integration.sh (hunk 4 above)
Comment (clebs) [reply prow/scripts/cluster-integration/kyma-gardener-integration.sh]: Done thanks!

File: prow/scripts/cluster-integration/kyma-gardener-integration.sh (hunk 5 above)
Comment (clebs) [reply prow/scripts/cluster-integration/kyma-gardener-integration.sh]: Done thanks!

File: prow/scripts/cluster-integration/kyma-gardener-integration.sh (hunk 6 above)
Comment (clebs) [reply prow/scripts/cluster-integration/kyma-gardener-integration.sh]: Done thanks!

**Pull request merged**